### DISCUSSION TITLE
Deploy preview comments on Storybook iframe

### DISCUSSION THREAD

**Discussion by robphoenix** [vercel/community#1123] [Help]
Hi, we've just started trying out Vercel deployments with our component library.
The [deploy preview comments functionality](https://vercel.com/docs/concepts/deployments/comments) isn't working with Storybook, I assume because Storybook is using iframes to display stories. Is this something that Vercel is aware of, and working on?
Thanks.

**Comment by amyegan** [comment]
Thanks for reaching out @robphoenix! I passed this along to the team and we'll update here when we know more. Do you have an example deployment or [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that I can share with the team for their investigation?

**Reply by robphoenix** [reply]
Hi @amyegan thanks for passing this along, I've set up [a repo](https://github.com/robphoenix/storybook-vercel-comments-example/pull/1), and there is [an example storybook](https://storybook-vercel-comments-example-git-comments-rphoenix.vercel.app/?path=/story/example-button--primary) you should be able to comment on (let me know if I need to grant anyone access). In storybook you can comment on the side panel, but not on the canvas panel (which is an iframe), for instance if you navigate to the primary button story and try to comment on the actual button, then nothing happens. If you open [the story in it's own tab](https://storybook-vercel-comments-example-git-comments-rphoenix.vercel.app/iframe.html?args=&id=example-button--primary&viewMode=story) (so, not within an iframe) you have to log in again, but you are able to comment.

**Reply by amyegan** [reply]
Thanks for sharing this, Rob! I asked the engineers, and they are working on a solution now. We'll keep sharing updates here as more info becomes available. Thanks for reporting this!

**Reply by robphoenix** [reply]
Awesome, thanks Amy 🙇🏻

**Reply by kylemh** [reply]
It's definitely because of iFrames. You can comment on the navigation, but not the actual documentation aspect of Storybook apps.

**Comment by anthonypaparella** [comment]
@amyegan Any update on this? Our team uses SB as our primary tool to view and approve UI components in our workflow. Storybook iframe still prevents us from being able to utilize Vercel preview comments feature.
